### PR TITLE
Strong matching iOS11 [PLATFORM-542]

### PR DESCRIPTION
**Pull request by parthkalavadia**
This is just a POC for strong matching in iOS 11 where it illustrates the user experience. Do not merge this PR.

### CODE REVIEW

**Comment by BranchMacMini** [comment]
Result of [Integration 1](https://stlt.herokuapp.com/v1/xcs_deeplink/qabot.stage.branch.io/cf6883a0f9e873dc2469e7db0a4d5699/cf6883a0f9e873dc2469e7db0a4d64d6)
---
*Duration*: 11 minutes and 34 seconds
*Result*: All 220 tests passed, but please **fix 1 warning**.
*Test Coverage*: 63%

**Comment by BranchMacMini** [comment]
Result of [Integration 2](https://stlt.herokuapp.com/v1/xcs_deeplink/qabot.stage.branch.io/cf6883a0f9e873dc2469e7db0a4d5699/cf6883a0f9e873dc2469e7db0ac88fde)
---
*Duration*: 11 minutes and 12 seconds
*Result*: All 220 tests passed, but please **fix 1 warning**.
*Test Coverage*: 63%

**Comment by BranchMacMini** [comment]
Result of [Integration 3](https://stlt.herokuapp.com/v1/xcs_deeplink/qabot.stage.branch.io/cf6883a0f9e873dc2469e7db0a4d5699/cf6883a0f9e873dc2469e7db0af37e96)
---
*Duration*: 12 minutes and 17 seconds
*Result*: **Build failed 1 test** out of 220

**Comment by BranchMacMini** [comment]
Result of [Integration 4](https://stlt.herokuapp.com/v1/xcs_deeplink/qabot.stage.branch.io/cf6883a0f9e873dc2469e7db0a4d5699/22338d2b4844ebc5791d27cef19fe974)
---
*Duration*: 10 minutes and 47 seconds
*Result*: All 220 tests passed, but please **fix 1 warning**.
*Test Coverage*: 63%

**Comment by aaustin** [comment]
High level - I think this should work, as long as we update the API and link service to plug into this code. A couple comments:
1. Please add some comment describing the additional code and its functionality
2. We'll need extensive crash testing on the update to the session params. This can cause trouble if there are reads in progress to that dictionary simultaneously while the update is occurring.

**Comment by BranchMacMini** [comment]
Result of [Integration 6](https://stlt.herokuapp.com/v1/xcs_deeplink/qabot.stage.branch.io/cf6883a0f9e873dc2469e7db0a4d5699/429944bccefe2b75e046418be30361a0)
---
*Duration*: 11 minutes and 23 seconds
*Result*: All 223 tests passed, but please **fix 1 warning**.
*Test Coverage*: 62%

File: Branch-SDK/Branch-SDK/Branch.m [hunk 1]
```diff
@@ -760,7 +761,48 @@ - (BOOL)continueUserActivity:(NSUserActivity *)userActivity {
...
+        NSArray<BNCKeyValue*> *queryItems = [BNCEncodingUtils queryItems:userActivity.webpageURL];
+        NSMutableDictionary *queryParams = [NSMutableDictionary new];
+        for (BNCKeyValue*item in queryItems) {
+            queryParams[item.key] = item;
+        }
+        
+        /*Following condition would check to see if the user deeplinked back after checking the
+         *match_guarantee from safari on iOS 11.
+         */
+        if (queryParams[@"$force_match"] &&
```
Comment (ethanneff) [inline Branch-SDK/Branch-SDK/Branch.m]: should this be `$force_strong_match`?

Comment (ethanneff) [reply Branch-SDK/Branch-SDK/Branch.m]: answered on call, `$force_match` is an internal value

**Comment by BranchMacMini** [comment]
Result of [Integration 7](https://stlt.herokuapp.com/v1/xcs_deeplink/qabot.stage.branch.io/cf6883a0f9e873dc2469e7db0a4d5699/4ae8ddcede172c40459bfe8c5fa678fc)
---
*Duration*: 10 minutes and 39 seconds
*Result*: **Build failed 1 test** out of 223

**Comment by E-B-Smith** [comment]
This is blocked on a back-end change still.

**Pull request closed without merging**